### ISSUE TITLE
VMware host module build failure under Pop!_OS 21.10

### ISSUE THREAD

**Issue by swvr** [pop-os/pop#2135]
**Distribution:**
```
$ cat /etc/os-release
NAME="Pop!_OS"
VERSION="21.10"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 21.10"
VERSION_ID="21.10"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=impish
UBUNTU_CODENAME=impish
LOGO=distributor-logo-pop-os
``` 
 
**Related Application and/or Package Version:**
```
$ apt policy gcc
gcc:
  Installed: 4:11.2.0-1ubuntu1
  Candidate: 4:11.2.0-1ubuntu1
  Version table:
 *** 4:11.2.0-1ubuntu1 500
        500 http://us.archive.ubuntu.com/ubuntu impish/main amd64 Packages
        100 /var/lib/dpkg/status
```

**Issue/Bug Description:**
After upgrading from Pop!_OS 21.04 to 21.10, VMware workstation host modules (vmware-host-modules) need to be recompiled but fail to compile with the error `expression in static assertion is not constant`.

The following issue has the complete error log: https://github.com/mkubecek/vmware-host-modules/issues/109

This appears to be specific to Pop!_OS. Even though it's using the same version of gcc as Ubuntu 21.10, this issue does not occur following a fresh install of Ubuntu 21.10 and neither does the issue occur in Fedora 35. As noted in the issue above, the maintainer has compiled these modules using several versions of gcc under openSUSE without error.

**Steps to reproduce:**
```
git clone https://github.com/mkubecek/vmware-host-modules.git
cd vmware-host-modules
git checkout workstation-16.2.1
make
```

**Expected behavior:**
Compilation of these modules should not fail. They have been compiled using other distros using several versions of gcc and there is no issue.

**Comment by mika314** [comment]
Did you get any workaround? I just updated Pop OS and getting these errors.

**Comment by swvr** [comment]
Yes, you'll find in this [thread](https://github.com/mkubecek/vmware-host-modules/issues/109) a couple of workarounds. The maintainer's patched branch is [here](https://github.com/mkubecek/vmware-host-modules/commits/tmp/workstation-16.2.1-k5.16).
